Misc/NEWS.d/next/Library/2020-10-17-14-46-16.bpo-42060.du-IYq.rst

```diff
@@ -0,0 +1 @@
+Replace `asserts` on HTTPResponse and HTTPConnection with `if...raises` so that the conditional checks work while running with the optimization flag switched on
```

Comment:
reST files/documentation shouldn't exceed 80 characters according to PEP 12 (although I can't tell if it's GitHub showing me the long lines or it really exceeds that, sorry if it isn't). Also reST requires two backticks for code, not one like markdown :).

 ```suggestion
Replace ``assert`` in :class:`http.client.HTTPResponse` and :class:`http.client.HTTPConnection` with ``raise`` 
so that conditional checks work while running with Python's runtime optimization
 ``-O`` flag enabled.
```